app/src/main/java/com/concordia/mcga/utilities/pathfinding/PathFinderTile.java

```diff
@@ -1,36 +1,50 @@
 package com.concordia.mcga.utilities.pathfinding;
 
-import android.util.Log;
+import android.support.annotation.NonNull;
 
-import org.apache.commons.lang3.builder.EqualsBuilder;
-import org.apache.commons.lang3.builder.HashCodeBuilder;
-import org.json.JSONException;
-import org.json.JSONObject;
+/**
+ * Object acts as a wrapper for {@link IndoorMapTile}. Holds
+ * additional information about the tiles that are specifically used
+ * by the pathfinder algorithms
+ */
+public class PathFinderTile implements Comparable<PathFinderTile> {
 
-public class PathFinderTile {
+    enum Type {START, DESTINATION}
 
-    public static final PathFinderTile MAX_COST = new PathFinderTile(Integer.MIN_VALUE, Integer.MIN_VALUE);
+    public static final PathFinderTile MAX_COST = new PathFinderTile();
 
     static {
         MAX_COST.setDistFromEnd(Integer.MAX_VALUE);
     }
 
+    private IndoorMapTile indoorMapTile;
     private Type tileType;
     private int distFromStart;
     private int distFromEnd;
-    private int coordinateX;
-    private int coordinateY;
     private PathFinderTile parent;
 
-    public PathFinderTile(int x, int y) {
-        this.coordinateX = x;
-        this.coordinateY = y;
+    private PathFinderTile(){}
+
+    public PathFinderTile(IndoorMapTile indoorMapTile) {
+        this.indoorMapTile = indoorMapTile;
+    }
+
+    public Type getTileType() {
+        return tileType;
     }
 
     public void setTileType(Type tileType) {
         this.tileType = tileType;
     }
 
+    public IndoorMapTile getIndoorMapTile() {
+        return indoorMapTile;
+    }
+
+    public void setIndoorMapTile(IndoorMapTile indoorMapTile) {
+        this.indoorMapTile = indoorMapTile;
+    }
+
     public int getCost() {
```

Comment:
Incorrect. Your initial assumption that getCost() returns a constant distance is false. It has 2 components: distFromEnd + distFromStart.

distFromEnd follow the heuristic that deltaX + deltaY is a correct approximation. This part will be constant for each tile.

distFromStart, however, does vary as the algorithm changes. The heuristic is not applied to this variable. Rather it is calculated by finding the shortest path between the node and the start node and current node given a path. This occurs in SingleMapPathfinder#nextIteration(), where it calls calculateDistFromStart().

There is no bug here as the costs are correctly compared in the compareTo.